File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int mini = prices[0];
        int profit = 0;
        int i = 1;
        while(i<prices.size()){
            profit = max(profit,(prices[i]-mini));
            mini = min(mini,prices[i]);
            i++;
        }
        return profit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        mini = prices[0]
        profit =0
        for i in range(1,len(prices)):
            profit = max(profit,prices[i]-mini)
            mini=min(mini,prices[i])

        return profit
        